src/components/LiquidityMining/Create/Steps/PairAndReward/AssetSelector/index.tsx

```diff
@@ -1,77 +1,249 @@
+      return 'Insufficient balance'
+    } else if (approvalState === ApprovalState.APPROVED && rewardMemo && rewardMemo.greaterThan('0')) {
+      return 'Approved'
+    } else {
+      return 'Approve'
+    }
+  }, [approvalState, account, userBalance, rewardMemo])
+
+  useEffect(() => {
+    if (setRewardsObject && rewardMemo && userBalance) {
+      setRewardsObject({
+        type: ActionType.APPROVALS_CHANGE,
+        payload: {
+          index: index,
+          approval:
+            approvalState === ApprovalState.PENDING
+              ? ApprovalState.NOT_APPROVED
+              : rewardMemo.greaterThan('0') &&
+                userBalance.greaterThan(rewardMemo) &&
+                approvalState === ApprovalState.APPROVED
+              ? ApprovalState.APPROVED
+              : (rewardMemo.greaterThan('0') && userBalance?.lessThan(rewardMemo)) ||
+                approvalState === ApprovalState.NOT_APPROVED
+              ? ApprovalState.NOT_APPROVED
+              : ApprovalState.UNKNOWN,
+        },
+      })
+    }
+
+    setAreButtonsDisabled(!!(!account || !rewardMemo || (userBalance && userBalance.lessThan(rewardMemo))))
+  }, [account, rewardMemo, userBalance, setRewardsObject, index, approvalState])
 
   useEffect(() => {
     if (currency0 && currency1) {
+      setTokenName('LP PAIR')
       setAssetTitle(`${unwrappedToken(currency0)?.symbol}/${unwrappedToken(currency1)?.symbol}`)
     } else if (currency0) {
+      setTokenName(unwrappedToken(currency0)?.name)
       setAssetTitle(unwrappedToken(currency0)?.symbol || null)
+    } else {
+      setTokenName(undefined)
+      setAssetTitle(`SELECT ${campaingType === CampaignType.TOKEN ? 'TOKEN' : 'PAIR'}`)
     }
-  }, [currency0, currency1])
+  }, [currency0, currency1, campaingType])
+
+  const handleDismiss = () => {
+    setAssetTitle(null)
+    if (onResetCurrency) onResetCurrency()
+    setTokenName(undefined)
+  }
 
   return (
-    <Flex flexDirection="column">
-      <Box mb="16px">
-        <TYPE.small fontWeight="600" color="text4" letterSpacing="0.08em">
-          {title}
-        </TYPE.small>
-      </Box>
-      <Box>
-        <Card selectable onClick={onClick}>
-          <Flex height="26px" width="100%" justifyContent="center" alignItems="center">
-            <Box mr={currency0 ? '8px' : '0'}>
-              <AssetLogo currency0={currency0} currency1={currency1} />
-            </Box>
-            <Box>
-              <TYPE.body
-                lineHeight="20px"
-                color="white"
-                data-testid={title.toLocaleLowerCase().replace(' ', '-') + '-select'}
+    <Flex flexDirection={'column'} key={index}>
+      <SmoothGradientCard
+        selectable
+        isToken={true}
+        active={currency0 !== undefined}
+        paddingBottom={
+          isReward && index === 0 && currency0 === undefined ? '45px' : isReward ? '30px' : '28px' + '!important'
```

Comment:
I think we are checking for `isReward` twice here. Also we shouldn't use `!important`.

Reply:
will remove !important thats some leftover...So its not double checking its first checking if its reward and is first one in line then other is checking if its reward and the third one is if its not reward...Since they are have different text positioning...